src/combinatorics/hall/default.lean

```diff
@@ -0,0 +1,235 @@
+/-
+Copyright (c) 2021 Alena Gusakov, Bhavik Mehta, Kyle Miller. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alena Gusakov, Bhavik Mehta, Kyle Miller
+-/
+import combinatorics.hall.basic
+import topology.category.Top.limits
+
+/-!
+# Hall's Marriage Theorem
```

Comment:
I don't think anything should go in `.default` files. Could that file be `combinatorics.hall.infinite` instead?

Reply:
I decided to put the infinite version in `combinatorics.hall.basic` and the finite one in `combinatorics.hall.finite`

Reply:
Urf, now that's weird that `.basic` imports another file. Any other idea?